### PR TITLE
Brian

### PR DESCRIPTION
**Pull request by bmduffy**
Hi Nic,

I've got the some results iterators and mocking working for scraping the results page. There are different types of parsers, depending on the offer, i.e.  new homes, for sale, for rent etc. I want to work on those next. Those classes will get the data into the form it needs to be in to be added as records in the redis data store.

Brian

### CODE REVIEW

File: daftpunk/results.py [hunk 1]
```diff
@@ -149,17 +151,12 @@ def get_page_url(self, offset):
...
-        stuff    = None
+    def get_response(self, url):
+        return requests.get(url)
 
-        if self.mode == "soup":
-            stuff = bs4.BeautifulSoup(response.content, "html.parser")
-        elif self.mode == "response":
-            stuff = response
+    def get_soup(self, url):
+        return bs4.BeautifulSoup(requests.get(url), "html.parser")
```
Comment (nicr9) [inline daftpunk/results.py]: Maybe this should be calling `self.get_response(...)` instead of `requests.get(...)`

File: daftpunk/results.py [hunk 2]
```diff
@@ -32,60 +32,76 @@ def next(self):
...
-        self.get_offest = offset
-
     def __iter__(self):
         return self
 
-    def to_return(self, offset, url, page):
-        
-        data = []
+    def next(self):
+        pass
```
Comment (nicr9) [inline daftpunk/results.py]: Could you change this to `raise StopIteration()` here instead of `pass`.

**Review by nicr9 — CHANGES_REQUESTED** [review]
Just two suggested changes. Other than that it looks good!

File: daftpunk/results.py (hunk 1 above)
Comment (bmduffy) [reply daftpunk/results.py]: Makes sense, I forgot to access the content byte data here to

self.get_response(url).content

File: daftpunk/results.py (hunk 2 above)
Comment (bmduffy) [reply daftpunk/results.py]: Good point, changed this

**Comment by nicr9** [comment]
Thanks!

**Pull request merged**